raw_socket_listener/listener.go

```diff
@@ -397,7 +397,12 @@ func (t *Listener) readPcap() {
 
 					srcIP = data[12:16]
 					dstIP = data[16:20]
-					data = data[ihl*4:]
+					if len(packet.Data()) < 64 && decoder == layers.LinkTypeEthernet { // beware the runt frame https://en.wikipedia.org/wiki/Ethernet_frame#Runt_frames
+						sliceLen := len(packet.Data()) - (len(packet.Data()) - int(binary.BigEndian.Uint16(data[2:4])) - of) - of
```

Comment:
So we are trying to calculate the actual end of the data packet (E) that is in a padded envelope.  IL is IP Envelope length. So
E = IL + pad
IL = E - pad

So `data = data[ihl * 4: int(binary.BigEndian.Uint16(data[2:4]))]`